src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -288,9 +288,13 @@ trait LowPriorityConfigConvertImplicits {
   }
 
   implicit val finiteDurationConfigConvert: ConfigConvert[FiniteDuration] = {
-    val fromString: String => FiniteDuration = (s: String) => DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]]).get match {
-      case v if v.isFinite() => Duration(v.length, v.unit)
-      case _ => throw new IllegalArgumentException(s"Couldn't parse '$s' into a finite duration because it's infinite.")
+    val fromString: String => FiniteDuration = { (s: String) =>
+      DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]])
+        .flatMap {
+          case d: FiniteDuration => Success(d)
+          case _ => Failure(new IllegalArgumentException(s"Couldn't parse '$s' into a FiniteDuration because it's infinite."))
+        }
+        .get
```

Comment:
@melrief I'm fine skipping the `Unsafe` if you don't see a reason to have them. 

In general the `Unsafe` are more convenient to call for most usages. For example

```
  implicit val readURL = stringConvert[URL](new URL(_), _.toString)
```
becomes
```
  implicit val readURL = stringConvert[URL](x => Try(new URL(x)), x => Try(x.toString))
```

I think there ultimately has be to a `Try` at the bottom of the call stack where all of these methods are called to avoid an unintentional exception from escaping, even if the lambda's type signature would suggest that shouldn't happen.